tests/controls/maskededittest.cpp

```diff
@@ -0,0 +1,149 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        tests/controls/maskededittest.cpp
+// Purpose:     wxMaskedEdit controls unit test
+// Author:      Manuel Martin
+// Created:     2012-07-30
+// RCS-ID:      $Id:
+// Copyright:   (c)
+///////////////////////////////////////////////////////////////////////////////
+
+#include "testprec.h"
+
+#if wxUSE_MASKED_EDIT
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/app.h"
+#endif // WX_PRECOMP
+
+#include "wx/maskededit.h"
+
+#include "testableframe.h"
+#include "wx/uiaction.h"
+
+class MaskedEditTestCase : public CppUnit::TestCase
+{
+public:
+    MaskedEditTestCase() { }
+
+    virtual void setUp() wxOVERRIDE
+    {
+        CreateControl();
+    }
+
+    virtual void tearDown() wxOVERRIDE
+    {
+        wxTheApp->GetTopWindow()->DestroyChildren();
+    }
+
+private:
+    CPPUNIT_TEST_SUITE( MaskedEditTestCase );
+    CPPUNIT_TEST( TestSet );
+    WXUISIM_TEST( TestUI );
+    CPPUNIT_TEST( TestFunc );
+    CPPUNIT_TEST_SUITE_END();
+
+    void TestSet();
+#if wxUSE_UIACTIONSIMULATOR
+    void TestUI();
+#endif
+    void TestFunc();
+    void CreateControl();
+    wxMaskedEditText* m_editTx;
+
+    DECLARE_NO_COPY_CLASS(MaskedEditTestCase)
+};
+
+// register in the unnamed registry so that these tests are run by default
+CPPUNIT_TEST_SUITE_REGISTRATION( MaskedEditTestCase );
+
+// also include in its own registry so that these tests can be run alone
+CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( MaskedEditTestCase, "MaskedEditTestCase" );
+
+void MaskedEditTestCase::CreateControl()
+{
+    m_editTx = new wxMaskedEditText(wxTheApp->GetTopWindow(), wxID_ANY);
+    m_editTx->SetMask("=a>A^-###++xx.#{4}");
+    // test empty value
+    CPPUNIT_ASSERT_EQUAL( "=  -   ++  .    ", m_editTx->GetValue() );
```

Comment:
We shouldn't have these tests in `CreateControl()` which is called at the beginning of each test as this means that we execute them many times instead of just once.